include/swift/Reflection/ReflectionContext.h

```diff
@@ -1182,14 +1182,11 @@ class ReflectionContext
     if (!DescriptorAddress)
       return false;
 
-    auto DescriptorBytes =
-        getReader().readBytes(RemoteAddress(DescriptorAddress),
-                              sizeof(TargetTypeContextDescriptor<Runtime>));
-    if (!DescriptorBytes)
+    auto Descriptor = readObj<TargetTypeContextDescriptor<Runtime>>(DescriptorAddress);
+    if (!Descriptor)
+      return false;
+    if (Descriptor->getKind() != ContextDescriptorKind::Class)
       return false;
-    auto Descriptor =
-        reinterpret_cast<const TargetTypeContextDescriptor<Runtime> *>(
-            DescriptorBytes.get());
     return Descriptor->getTypeContextDescriptorFlags().class_isActor();
```

Comment:
What do you think of rewriting this as:

```suggestion
if (auto *Descriptor = readObj<TargetTypeContextContextDescriptor<Runtime>>(DescriptorAddress))
  if (Descriptor->getKind() == ContextDescriptorKind::Class)
    return Descriptor->getTypeContextDescriptroFlags().class_isActor();
return false;
```

Reply:
I think the early bailout is a little more LLVMy (not that I'm a huge fan of LLVM style, but when in Rome...) and I suspect I'll be adding more logic here soon to do better detection, which would fit better in an early-bailout pattern.

Reply:
LLVM does both, particularly when it limits the scope of the variable.  If you have plans of extending this further where you are going to use the `Descriptor` outside of just that scope, it makes sense to keep the current structure.